networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/gossip/config/GossipScoringConfig.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.p2p.gossip.config;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+/** Gossip scoring config. Contains peer scoring and topic scoring configs. */
+public class GossipScoringConfig {
```

Comment:
I went back and forth on this - still not 100% sure which way is better.  But I ended up making these builders in an attempt to follow existing config patterns, and to try to keep jvm-libp2p-specific dependencies contained within the `libp2p` package.  The `libp2p` package is really meant to be in a separate higher-level module that depends on the core `p2p` interfaces.  🤷‍♀️

Reply:
> The libp2p package is really meant to be in a separate higher-level module that depends on the core p2p interfaces. 🤷‍♀️

Yep I think I see what you mean here. However I generally try to avoid over-abstraction when it's not yet planned to be used. I.e. its unlikely that we will use anything other than `jvm libp2p` for out `p2p` so abstracting over this from one side could keep the code and interfaces cleaner, but from the other side will cost you more in maintenance. 

TBH I like our `p2p` and `eth2` segregation from architectural point of view, but they are becoming more tightly coupled and making changes becoming more complicated

Reply:
Opened a follow-up ticket to look into simplifying: https://github.com/ConsenSys/teku/issues/3573